source/Ship.cpp

```diff
@@ -220,6 +221,16 @@ namespace {
 
 
 
+// Determine if this bay can hold a ship of this category.
+bool Ship::Bay::CanContain(const std::string &category) const
+{
+	if(bayType)
+		return bayType->Contains(category);
+	return name == category;
```

Comment:
```suggestion
	return bayType ? bayType->Contains(category) : name == category;
```
this would be my personal preference, do as you like